src/ray/raylet/scheduling/policy/scheduling_policy.h

```diff
@@ -94,9 +97,47 @@ class ISchedulingPolicy {
   virtual SchedulingResult Schedule(
       const std::vector<const ResourceRequest *> &resource_request_list,
       SchedulingOptions options,
-      SchedulingContext *context) {
-    return SchedulingResult();
+      SchedulingContext *context = nullptr) = 0;
+};
+
+class ISingleSchedulingPolicy : public ISchedulingPolicy {
+ public:
+  /// \param resource_request: The resource request we're attempting to schedule.
+  /// \param options: scheduling options.
+  ///
+  /// \return NodeID::Nil() if the task is unfeasible, otherwise the node id
+  /// to schedule on.
+  virtual scheduling::NodeID Schedule(const ResourceRequest &resource_request,
+                                      SchedulingOptions options,
+                                      SchedulingContext *context = nullptr) = 0;
+
+  SchedulingResult Schedule(
+      const std::vector<const ResourceRequest *> &resource_request_list,
+      SchedulingOptions options,
+      SchedulingContext *context = nullptr) override {
+    size_t success_count = 0;
+    std::vector<scheduling::NodeID> selected_nodes;
+    selected_nodes.reserve(resource_request_list.size());
+    for (auto &resource_request : resource_request_list) {
```

Comment:
I don't think we really support scheduling multiple requests at the same time. In order for that to work, the resource view needs to be updated after each scheduling decision.

I feel we are trying to pack different things into a single interface: SchedulingContext is for pg, SchedulingOptions is for tasks/actors, multiple requests for pg, single request for tasks/actors. I feel we should think more about the interface. @scv119

Reply:
I agree this is not ideal today. My hope is that scheduling policy layer becomes PG agnostic but only worry about schedule a single or a batch of resources.

Reply:
+1. I believe this is the code we have used though. Maybe we should have a follow up to clean up the interface? Agreed pg scheduling is just a batch scheduling, and it can be pg agonistic

Reply:
I think the batch scheduling interface will be useful for tasks and actors as well.
For example, when many tasks/actors are submitted all at once and pending on the raylet/GCS side, we can pass multiple  of them to the scheduler. This will potentially improve the efficiency and effectiveness.
So I'd like to have the callers of this interface pass many requests when possible. Then it's up to the implementation to decide whether to do single or batch scheduling.